FBKVOController/FBKVOController.m

```diff
@@ -275,12 +288,20 @@ - (void)unobserve:(id)object info:(_FBKVOInfo *)info
   if (nil == info) {
     return;
   }
-  
+
+  if (!info->_registered) {
+      // this could happen when `NSKeyValueObservingOptionInitial` is one of the NSKeyValueObservingOptions.
+      // we cannot unobserve them right now because they are not technially observers yet.
+      // instead, we set a flag so they can be unobserved later, after the `addObserver` call returns.
+      info->_unregisteredBeforeRegistered = YES;
+      return;
```

Comment:
Thanks for the feedback! Although I am not 100% sure what you mean here. In my current implementation, if we are unobserving the observer inside the block with `.Initial` option set, `- unobserve:info(s):` method will actually got called twice: (1) inside the block, and (2) from `- observe:info:` method.

Although it will left the observation info in the context at first, `info` will be removed the second time we attempt to unobserve it. I have verified that `_infos.count == 0` at the time the call to `- observe:info:` returns.

However, I think the implementation could become simpler if we implement it as you suggested - I think we still need to call `- unobserve:` twice, but removing the same `info` object twice from the NSHashTable seems harmless, so I will give it a try.